File: clients/java/examples/collections-find-one-lexical.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.options.CollectionFindOneOptions;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Filters;
import com.datastax.astra.client.core.query.Sort;
import java.util.Optional;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Find a document
    CollectionFindOneOptions options =
        new CollectionFindOneOptions().sort(Sort.lexical("tree hill grassy"));
    Optional<Document> result = collection.findOne(Filters.match("tree hill"), options);
    System.out.println(result);
  }
}

File: clients/python/examples/collections-find-one-lexical.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Find a document
result = collection.find_one(
    {"$lexical": {"$match": "tree hill"}},
    sort={"$lexical": "tree hill grassy"},
)

print(result)
